### ISSUE TITLE
Module not found: Error: Can't resolve 'v8'

### ISSUE THREAD

**Issue by astonishqft** [babel/babel#13067]
## Bug Report

- [ ] I would like to work on a fix!

```
* v8 in ./node_modules/@babel/core/lib/transformation/util/clone-deep.js

To install it, you can run: npm install --save v8
Child
    
    ERROR in ./node_modules/@babel/core/lib/transformation/util/clone-deep.js
    Module not found: Error: Can't resolve 'v8' in '/Users/qifutao/Documents/workspace/dxp-flow/frontend/node_modules/@babel/core/lib/transformation/util'
     @ ./node_modules/@babel/core/lib/transformation/util/clone-deep.js 16:9-22
     @ ./node_modules/@babel/core/lib/transformation/normalize-file.js
     @ ./node_modules/@babel/core/lib/transformation/index.js
     @ ./node_modules/@babel/core/lib/transform-ast.js
     @ ./node_modules/@babel/core/lib/index.js
     @ ./node_modules/@whalecloud/page-builder/utils/transformJS.js
     @ ./node_modules/@whalecloud/page-builder/render/components/RenderForm.js
     @ ./node_modules/@whalecloud/page-builder/render/components/FormRender.js
     @ ./src/pages/approveList/index.tsx
     @ ./src/.fishx/config/routes.js
     @ ./src/.fishx/router.tsx
     @ ./src/.fishx/index.tsx
     @ multi ./src/.fishx/index.tsx
```

**Environment**

```

```
- Babel version(s): v7.13.13
- Node/npm version: Node 12
- OS: [e.g. macOS 10.15.4]
- Monorepo: yes
- How you are using Babel: `webpack`

**Possible Solution**

**Additional context**

**Comment by astonishqft** [comment]
It looks like someone rewrite `babel-core/src/transformation/util/clone-deep.js`,

```js
import v8 from "v8";
import cloneDeep from "./clone-deep-browser";

export default function (value) {
  if (v8.deserialize && v8.serialize) {
    return v8.deserialize(v8.serialize(value));
  }
  return cloneDeep(value);
}
```

however, the package of v8 is not specified in package.json.

```json
{
  "name": "@babel/core",
  "version": "7.13.13",
  "description": "Babel compiler core.",
  "main": "lib/index.js",
  "author": "Sebastian McKenzie <sebmck@gmail.com>",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/babel/babel.git",
    "directory": "packages/babel-core"
  },
  "homepage": "https://babel.dev/docs/en/next/babel-core",
  "bugs": "https://github.com/babel/babel/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22pkg%3A%20core%22+is%3Aopen",
  "keywords": [
    "6to5",
    "babel",
    "classes",
    "const",
    "es6",
    "harmony",
    "let",
    "modules",
    "transpile",
    "transpiler",
    "var",
    "babel-core",
    "compiler"
  ],
  "engines": {
    "node": ">=6.9.0"
  },
  "funding": {
    "type": "opencollective",
    "url": "https://opencollective.com/babel"
  },
  "browser": {
    "./lib/config/files/index.js": "./lib/config/files/index-browser.js",
    "./lib/config/resolve-targets.js": "./lib/config/resolve-targets-browser.js",
    "./lib/transform-file.js": "./lib/transform-file-browser.js",
    "./src/config/files/index.js": "./src/config/files/index-browser.js",
    "./src/config/resolve-targets.js": "./src/config/resolve-targets-browser.js",
    "./src/transform-file.js": "./src/transform-file-browser.js",
    "./src/transformation/util/clone-deep.js": "./src/transformation/util/clone-deep-browser.js"
  },
  "dependencies": {
    "@babel/code-frame": "workspace:^7.12.13",
    "@babel/generator": "workspace:^7.13.9",
    "@babel/helper-compilation-targets": "workspace:^7.13.13",
    "@babel/helper-module-transforms": "workspace:^7.13.12",
    "@babel/helpers": "workspace:^7.13.10",
    "@babel/parser": "workspace:^7.13.13",
    "@babel/template": "workspace:^7.12.13",
    "@babel/traverse": "workspace:^7.13.13",
    "@babel/types": "workspace:^7.13.13",
    "convert-source-map": "^1.7.0",
    "debug": "^4.1.0",
    "escape-string-regexp": "condition:BABEL_8_BREAKING ? ^4.0.0 : ",
    "gensync": "^1.0.0-beta.2",
    "json5": "^2.1.2",
    "lodash": "^4.17.19",
    "semver": "condition:BABEL_8_BREAKING ? ^7.3.4 : ^6.3.0",
    "source-map": "^0.5.0"
  },
  "devDependencies": {
    "@babel/helper-transform-fixture-test-runner": "workspace:*"
  }
}
```

**Comment by JLHwung** [comment]
Did you specify "targets: web" on your webpack config? v8 is a node.js built in and thus not supposed to be used in Browsers. 

We have provided a JSON alternative for browsers, but webpack should be configured to respect the "browser" field.

**Comment by davedx** [comment]
Can confirm this breaks our build too, we're using react-scripts so I don't think it's a webpack misconfiguration on our side.

Error looks like this:

> react-scripts build

Creating an optimized production build...
Failed to compile.

...../node_modules/react-view/node_modules/@babel/core/lib/transformation/util/clone-deep.js

Cannot find module: 'v8'. Make sure this package is installed.

You can install this package by running: npm install v8.

**Comment by davedx** [comment]
Reproduction repo https://github.com/davedx/babel-v8-repro

Note: I first used create-react-app to make a new react-scripts project and when I ran yarn build, it worked fine. Puzzled, I then added 'react-view' (which we use in our project), and yarn build broke. At first I thought maybe it was react-view that was the issue. But looking into the package json of react-scripts and react-view, the main difference I see immediately is that react-scripts depends on "@babel/core": "7.12.3", and react-view has "@babel/core": "^7.13.8". I'm still investigating, will post more if I find it

**Comment by davedx** [comment]
On further investigation I am stumped as to why when running in a node environment, I get an error that v8 cannot be found. For example this module (trying to replicate how babel is pulling in v8) works:

test.mjs:

import v8 from "v8";

console.log('v8: ', v8)

node --experimental-modules test.mjs
(node:26809) ExperimentalWarning: The ESM module loader is experimental.
v8:  {
  cachedDataVersionTag: [Function: cachedDataVersionTag],
[snip]

There is some kind of voodoo going on here that I simply don't understand

**Comment by JLHwung** [comment]
Fixed in `@babel/core` 7.13.14. Thanks for everyone who reported this issue, created a reproduction repo and came up with a fix!
